tests/includes/wait-for.sh

```diff
@@ -78,10 +78,11 @@ active_condition() {
 	echo ".applications | select(.[\"$name\"] | .[\"application-status\"] | .current == \"active\") | keys[$app_index]"
 }
 
-# if an application is in error, we want to count it as an application which is not idle
-# not_idle_list lists all such applications.
-not_idle_list() {
-	echo '[.applications[] | select((.units[] | .["juju-status"].current != "idle") or (.units[] | .["workload-status"].current == "error"))] | length'
+# not_idle_count should be used where you expect an arbitrary number of applications' whose agent-status are not in idle state,
+# ideally applications in a bundle, this helps the tests to avoid being overly specific to a given no. of applications.
+# e.g. wait_for 0 "$(not_idle_count) | length" 1800
+not_idle_count() {
+	echo '[.applications[] | select((.units[] | .["juju-status"].current != "idle") or (.units[] | .["workload-status"].current == "error"))]'
```

Comment:
This function needs updating. Beforehand it was misleading because it claimed to return a list, but returned a count. Now it claims to return a count, but returns a list.

Reply:
Thx, I've renamed the function to not_idle_list.